### ISSUE TITLE
weird stopping at `openal-soft`

### ISSUE THREAD

**Issue by hooke007** [shinchiro/mpv-winbuild-cmake#164]
gcc11

ninja mpv

```
[103/477] Performing build step for 'openal-soft'
FAILED: packages/openal-soft-prefix/src/openal-soft-stamp/openal-soft-build /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft-stamp/openal-soft-build 
cd /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft-build && /usr/bin/cmake -P /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft-stamp/openal-soft-build-.cmake && /usr/bin/cmake -E touch /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft-stamp/openal-soft-build
CMake Error at /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft-stamp/openal-soft-build-.cmake:49 (message):
  Command failed: 1

   '/usr/bin/ninja'

  See also

    /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft-stamp/openal-soft-build-*.log
```

openal-soft-build-out.log

```
[1/3] cd /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft && /usr/bin/cmake -D GIT_EXECUTABLE=/usr/bin/git -D LIB_VERSION=1.21.1 -D LIB_VERSION_NUM=1,21,1,0 -D SRC=/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/version.h.in -D DST=/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft-build/version.h -P /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/version.cmake
[2/3] Building CXX object CMakeFiles/OpenAL.dir/alc/backends/pipewire.cpp.obj
FAILED: CMakeFiles/OpenAL.dir/alc/backends/pipewire.cpp.obj 
/home/hooke007/mpv64/install/bin/x86_64-w64-mingw32-g++ -DALC_API="" -DAL_ALEXT_PROTOTYPES -DAL_API="" -DAL_BUILD_LIBRARY -DAL_LIBTYPE_STATIC -DNOMINMAX -DRESTRICT=__restrict -D_WIN32 -D__USE_MINGW_ANSI_STDIO -I/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/include -I/usr/include/pipewire-0.3 -I/usr/include/spa-0.2 -I/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft-build -I/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft -I/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/common -O2 -g -D_DEBUG -Winline -Wunused -Wall -Wextra -Wshadow -Wconversion -Wcast-align -Wpedantic -Wold-style-cast -Wnon-virtual-dtor -Woverloaded-virtual -fno-math-errno -std=gnu++14 -MD -MT CMakeFiles/OpenAL.dir/alc/backends/pipewire.cpp.obj -MF CMakeFiles/OpenAL.dir/alc/backends/pipewire.cpp.obj.d -o CMakeFiles/OpenAL.dir/alc/backends/pipewire.cpp.obj -c /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/backends/pipewire.cpp
In file included from /usr/include/pipewire-0.3/pipewire/protocol.h:48,
                 from /usr/include/pipewire-0.3/pipewire/proxy.h:108,
                 from /usr/include/pipewire-0.3/pipewire/client.h:35,
                 from /usr/include/pipewire-0.3/pipewire/pipewire.h:35,
                 from /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/backends/pipewire.cpp:54:
/usr/include/pipewire-0.3/pipewire/utils.h:33:10: fatal error: sys/un.h: No such file or directory
   33 | #include <sys/un.h>
      |          ^~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
```

**Comment by hooke007** [comment]
windows doesn't use some backends built by openal-soft, possible to disable them ?

**Comment by shinchiro** [comment]
It run wells on my system. Pipewire codes shouldn't be compiled though

**Comment by hooke007** [comment]
well, full-uninstall the pipewire in my arch-kde could finally pass the building.
Eh, pipewire was default installed by plasma....
Hope it useful for those who use arch's desktop environment to build mpv.

openal-soft-build-out.log
```
[1/94] cd /home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft && /usr/bin/cmake -D GIT_EXECUTABLE=/usr/bin/git -D LIB_VERSION=1.21.1 -D LIB_VERSION_NUM=1,21,1,0 -D SRC=/home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/version.h.in -D DST=/home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft-build/version.h -P /home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/version.cmake
[2/94] Generating hrtf_default.h
[3/94] Building CXX object CMakeFiles/OpenAL.dir/common/alstring.cpp.obj
[4/94] Building CXX object CMakeFiles/OpenAL.dir/common/polyphase_resampler.cpp.obj
[5/94] Building CXX object CMakeFiles/OpenAL.dir/common/almalloc.cpp.obj
[6/94] Building CXX object CMakeFiles/OpenAL.dir/common/ringbuffer.cpp.obj
[7/94] Building CXX object CMakeFiles/OpenAL.dir/common/alcomplex.cpp.obj
[8/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/autowah.cpp.obj
[9/94] Building CXX object CMakeFiles/OpenAL.dir/common/threads.cpp.obj
[10/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/compressor.cpp.obj
[11/94] Building CXX object CMakeFiles/OpenAL.dir/common/strutils.cpp.obj
[12/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/chorus.cpp.obj
[13/94] Building CXX object CMakeFiles/OpenAL.dir/common/dynload.cpp.obj
[14/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/convolution.cpp.obj
[15/94] Building CXX object CMakeFiles/OpenAL.dir/common/alfstream.cpp.obj
[16/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/dedicated.cpp.obj
[17/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/distortion.cpp.obj
[18/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/echo.cpp.obj
[19/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/equalizer.cpp.obj
[20/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/pshifter.cpp.obj
[21/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/fshifter.cpp.obj
[22/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/null.cpp.obj
[23/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/modulator.cpp.obj
[24/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/vmorpher.cpp.obj
[25/94] Building CXX object CMakeFiles/OpenAL.dir/al/effects/reverb.cpp.obj
[26/94] Building CXX object CMakeFiles/OpenAL.dir/al/effect.cpp.obj
[27/94] Building CXX object CMakeFiles/OpenAL.dir/al/auxeffectslot.cpp.obj
[28/94] Building CXX object CMakeFiles/OpenAL.dir/al/buffer.cpp.obj
[29/94] Building CXX object CMakeFiles/OpenAL.dir/al/extension.cpp.obj
[30/94] Building CXX object CMakeFiles/OpenAL.dir/al/event.cpp.obj
[31/94] Building CXX object CMakeFiles/OpenAL.dir/al/listener.cpp.obj
[32/94] Building CXX object CMakeFiles/OpenAL.dir/al/error.cpp.obj
[33/94] Building CXX object CMakeFiles/OpenAL.dir/al/state.cpp.obj
[34/94] Building CXX object CMakeFiles/OpenAL.dir/al/filter.cpp.obj
[35/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effectslot.cpp.obj
[36/94] Building CXX object CMakeFiles/OpenAL.dir/alc/device.cpp.obj
[37/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/autowah.cpp.obj
[38/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/compressor.cpp.obj
[39/94] Building CXX object CMakeFiles/OpenAL.dir/alc/context.cpp.obj
[40/94] Building CXX object CMakeFiles/OpenAL.dir/alc/alu.cpp.obj
[41/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/dedicated.cpp.obj
[42/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/chorus.cpp.obj
[43/94] Building CXX object CMakeFiles/OpenAL.dir/alc/alconfig.cpp.obj
[44/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/null.cpp.obj
[45/94] Building CXX object CMakeFiles/OpenAL.dir/al/source.cpp.obj
[46/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/distortion.cpp.obj
[47/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/equalizer.cpp.obj
[48/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/convolution.cpp.obj
[49/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/echo.cpp.obj
[50/94] Building CXX object CMakeFiles/OpenAL.dir/alc/alc.cpp.obj
In file included from /home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/alc/alconfig.h:6,
                 from /home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/alc/device.h:14,
                 from /home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/al/auxeffectslot.h:11,
                 from /home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/alc/alc.cpp:62:
/home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/common/aloptional.h: In function 'ALCenum {anonymous}::UpdateDeviceParams(ALCdevice*, const int*)':
/home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/common/aloptional.h:313:78: warning: 'optlimit.al::optional<bool>::mStore.al::detail_::optional_storage<bool, true, true, true, true>::<anonymous>.al::detail_::optstore_helper<bool>::<anonymous>.al::detail_::optstore_base<bool, true>::<anonymous>.al::detail_::optstore_base<bool, true>::<unnamed union>::mDummy' may be used uninitialized in this function [-Wmaybe-uninitialized]
  313 |     { return bool{*this} ? **this : static_cast<T>(std::forward<U>(defval)); }
      |                                                                              ^
/home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/alc/alc.cpp:1536:24: note: 'optlimit.al::optional<bool>::mStore.al::detail_::optional_storage<bool, true, true, true, true>::<anonymous>.al::detail_::optstore_helper<bool>::<anonymous>.al::detail_::optstore_base<bool, true>::<anonymous>.al::detail_::optstore_base<bool, true>::<unnamed union>::mDummy' was declared here
 1536 |     al::optional<bool> optlimit{};
      |                        ^~~~~~~~
/home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/alc/alc.cpp:1693:13: warning: '*(unsigned char*)((char*)&optchans + offsetof(al::optional<DevFmtChannels>,al::optional<DevFmtChannels>::mStore.al::detail_::optional_storage<DevFmtChannels, true, true, true, true>::<unnamed>.al::detail_::optstore_helper<DevFmtChannels>::<unnamed>.al::detail_::optstore_base<DevFmtChannels, true>::<unnamed>))' may be used uninitialized in this function [-Wmaybe-uninitialized]
 1693 |             if(device->FmtChans == DevFmtAmbi3D)
      |             ^~
/home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/alc/alc.cpp:1696:39: warning: '*(unsigned char*)((char*)&optlayout + offsetof(al::optional<DevAmbiLayout>,al::optional<DevAmbiLayout>::mStore.al::detail_::optional_storage<DevAmbiLayout, true, true, true, true>::<unnamed>.al::detail_::optstore_helper<DevAmbiLayout>::<unnamed>.al::detail_::optstore_base<DevAmbiLayout, true>::<unnamed>))' may be used uninitialized in this function [-Wmaybe-uninitialized]
 1696 |                 device->mAmbiLayout = *optlayout;
      |                                       ^~~~~~~~~~
/home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/alc/alc.cpp:1697:36: warning: '*(unsigned char*)((char*)&optscale + offsetof(al::optional<DevAmbiScaling>,al::optional<DevAmbiScaling>::mStore.al::detail_::optional_storage<DevAmbiScaling, true, true, true, true>::<unnamed>.al::detail_::optstore_helper<DevAmbiScaling>::<unnamed>.al::detail_::optstore_base<DevAmbiScaling, true>::<unnamed>))' may be used uninitialized in this function [-Wmaybe-uninitialized]
 1697 |                 device->mAmbiScale = *optscale;
      |                 ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
/home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/alc/alc.cpp:1686:35: warning: '*(unsigned char*)((char*)&opthrtf + offsetof(al::optional<bool>,al::optional<bool>::mStore.al::detail_::optional_storage<bool, true, true, true, true>::<unnamed>.al::detail_::optstore_helper<bool>::<unnamed>.al::detail_::optstore_base<bool, true>::<unnamed>))' may be used uninitialized in this function [-Wmaybe-uninitialized]
 1686 |             stereomode = *opthrtf ? StereoEncoding::Hrtf : StereoEncoding::Normal;
      |                          ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[51/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/vmorpher.cpp.obj
[52/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/modulator.cpp.obj
[53/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/pshifter.cpp.obj
[54/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/fshifter.cpp.obj
[55/94] Building CXX object CMakeFiles/OpenAL.dir/alc/backends/loopback.cpp.obj
[56/94] Building CXX object CMakeFiles/OpenAL.dir/core/ambidefs.cpp.obj
[57/94] Building CXX object CMakeFiles/OpenAL.dir/alc/backends/null.cpp.obj
[58/94] Building CXX object CMakeFiles/OpenAL.dir/alc/backends/base.cpp.obj
[59/94] Building CXX object CMakeFiles/OpenAL.dir/core/bs2b.cpp.obj
[60/94] Building CXX object CMakeFiles/OpenAL.dir/core/buffer_storage.cpp.obj
[61/94] Building CXX object CMakeFiles/OpenAL.dir/core/bsinc_tables.cpp.obj
[62/94] Building CXX object CMakeFiles/OpenAL.dir/core/bformatdec.cpp.obj
[63/94] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/reverb.cpp.obj
[64/94] Building CXX object CMakeFiles/OpenAL.dir/core/devformat.cpp.obj
[65/94] Building CXX object CMakeFiles/OpenAL.dir/core/context.cpp.obj
[66/94] Building CXX object CMakeFiles/OpenAL.dir/core/device.cpp.obj
[67/94] Building CXX object CMakeFiles/OpenAL.dir/core/except.cpp.obj
[68/94] Building CXX object CMakeFiles/OpenAL.dir/alc/backends/dsound.cpp.obj
[69/94] Building CXX object CMakeFiles/OpenAL.dir/core/cpu_caps.cpp.obj
[70/94] Building CXX object CMakeFiles/OpenAL.dir/alc/panning.cpp.obj
In file included from /home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/alc/panning.cpp:50:
/home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/core/ambdec.h: In function 'void aluInitRenderer(ALCdevice*, int, al::optional<StereoEncoding>)':
/home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/core/ambdec.h:18:8: warning: inlining failed in call to 'AmbDecConf::~AmbDecConf() noexcept': call is unlikely and code size would grow [-Winline]
   18 | struct AmbDecConf {
      |        ^~~~~~~~~~
/home/hooke/mb64/packages/openal-soft-prefix/src/openal-soft/alc/panning.cpp:896:9: note: called from here
  896 |         };
      |         ^
[71/94] Building CXX object CMakeFiles/OpenAL.dir/core/fmt_traits.cpp.obj
[72/94] Building CXX object CMakeFiles/OpenAL.dir/alc/backends/wave.cpp.obj
[73/94] Building CXX object CMakeFiles/OpenAL.dir/core/filters/nfc.cpp.obj
[74/94] Building CXX object CMakeFiles/OpenAL.dir/core/filters/biquad.cpp.obj
[75/94] Building CXX object CMakeFiles/OpenAL.dir/core/converter.cpp.obj
[76/94] Building CXX object CMakeFiles/OpenAL.dir/alc/backends/winmm.cpp.obj
[77/94] Building CXX object CMakeFiles/OpenAL.dir/core/ambdec.cpp.obj
[78/94] Building CXX object CMakeFiles/OpenAL.dir/core/filters/splitter.cpp.obj
[79/94] Building CXX object CMakeFiles/OpenAL.dir/core/mixer.cpp.obj
[80/94] Building CXX object CMakeFiles/OpenAL.dir/core/fpu_ctrl.cpp.obj
[81/94] Building CXX object CMakeFiles/OpenAL.dir/core/mastering.cpp.obj
[82/94] Building CXX object CMakeFiles/OpenAL.dir/core/logging.cpp.obj
[83/94] Building CXX object CMakeFiles/OpenAL.dir/core/uiddefs.cpp.obj
[84/94] Building CXX object CMakeFiles/OpenAL.dir/core/mixer/mixer_sse3.cpp.obj
[85/94] Building CXX object CMakeFiles/OpenAL.dir/core/uhjfilter.cpp.obj
[86/94] Building CXX object CMakeFiles/OpenAL.dir/core/mixer/mixer_c.cpp.obj
[87/94] Building CXX object CMakeFiles/OpenAL.dir/core/mixer/mixer_sse.cpp.obj
[88/94] Building CXX object CMakeFiles/OpenAL.dir/alc/backends/wasapi.cpp.obj
[89/94] Building CXX object CMakeFiles/OpenAL.dir/core/mixer/mixer_sse41.cpp.obj
[90/94] Building CXX object CMakeFiles/OpenAL.dir/core/mixer/mixer_sse2.cpp.obj
[91/94] Building CXX object CMakeFiles/OpenAL.dir/core/voice.cpp.obj
[92/94] Building CXX object CMakeFiles/OpenAL.dir/core/helpers.cpp.obj
[93/94] Building CXX object CMakeFiles/OpenAL.dir/core/hrtf.cpp.obj
[94/94] Linking CXX static library libOpenAL32.a
```

**Comment by shinchiro** [comment]
Can you test adding `-DALSOFT_REQUIRE_PIPEWIRE=OFF` to configure line while having pipewire install? Check whether it sucessfully compile or not

**Comment by hooke007** [comment]
Nope.

```
[1/95] cd /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft && /usr/bin/cmake -D GIT_EXECUTABLE=/usr/bin/git -D LIB_VERSION=1.21.1 -D LIB_VERSION_NUM=1,21,1,0 -D SRC=/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/version.h.in -D DST=/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft-build/version.h -P /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/version.cmake
[2/95] Generating hrtf_default.h
[3/95] Building CXX object CMakeFiles/OpenAL.dir/common/alstring.cpp.obj
[4/95] Building CXX object CMakeFiles/OpenAL.dir/common/polyphase_resampler.cpp.obj
[5/95] Building CXX object CMakeFiles/OpenAL.dir/common/ringbuffer.cpp.obj
[6/95] Building CXX object CMakeFiles/OpenAL.dir/common/almalloc.cpp.obj
[7/95] Building CXX object CMakeFiles/OpenAL.dir/common/alcomplex.cpp.obj
[8/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/autowah.cpp.obj
[9/95] Building CXX object CMakeFiles/OpenAL.dir/common/strutils.cpp.obj
[10/95] Building CXX object CMakeFiles/OpenAL.dir/common/threads.cpp.obj
[11/95] Building CXX object CMakeFiles/OpenAL.dir/common/dynload.cpp.obj
[12/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/chorus.cpp.obj
[13/95] Building CXX object CMakeFiles/OpenAL.dir/common/alfstream.cpp.obj
[14/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/compressor.cpp.obj
[15/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/echo.cpp.obj
[16/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/convolution.cpp.obj
[17/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/dedicated.cpp.obj
[18/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/distortion.cpp.obj
[19/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/equalizer.cpp.obj
[20/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/modulator.cpp.obj
[21/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/fshifter.cpp.obj
[22/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/pshifter.cpp.obj
[23/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/null.cpp.obj
[24/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/reverb.cpp.obj
[25/95] Building CXX object CMakeFiles/OpenAL.dir/al/auxeffectslot.cpp.obj
[26/95] Building CXX object CMakeFiles/OpenAL.dir/al/effects/vmorpher.cpp.obj
[27/95] Building CXX object CMakeFiles/OpenAL.dir/al/effect.cpp.obj
[28/95] Building CXX object CMakeFiles/OpenAL.dir/al/buffer.cpp.obj
[29/95] Building CXX object CMakeFiles/OpenAL.dir/al/error.cpp.obj
[30/95] Building CXX object CMakeFiles/OpenAL.dir/al/extension.cpp.obj
[31/95] Building CXX object CMakeFiles/OpenAL.dir/al/event.cpp.obj
[32/95] Building CXX object CMakeFiles/OpenAL.dir/al/state.cpp.obj
[33/95] Building CXX object CMakeFiles/OpenAL.dir/al/listener.cpp.obj
[34/95] Building CXX object CMakeFiles/OpenAL.dir/al/filter.cpp.obj
[35/95] Building CXX object CMakeFiles/OpenAL.dir/alc/device.cpp.obj
[36/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effectslot.cpp.obj
[37/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/autowah.cpp.obj
[38/95] Building CXX object CMakeFiles/OpenAL.dir/alc/context.cpp.obj
[39/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/chorus.cpp.obj
[40/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/compressor.cpp.obj
[41/95] Building CXX object CMakeFiles/OpenAL.dir/alc/alu.cpp.obj
[42/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/dedicated.cpp.obj
[43/95] Building CXX object CMakeFiles/OpenAL.dir/al/source.cpp.obj
[44/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/null.cpp.obj
[45/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/equalizer.cpp.obj
[46/95] Building CXX object CMakeFiles/OpenAL.dir/alc/alconfig.cpp.obj
[47/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/distortion.cpp.obj
[48/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/echo.cpp.obj
[49/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/fshifter.cpp.obj
[50/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/modulator.cpp.obj
[51/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/convolution.cpp.obj
[52/95] Building CXX object CMakeFiles/OpenAL.dir/alc/backends/pipewire.cpp.obj
FAILED: CMakeFiles/OpenAL.dir/alc/backends/pipewire.cpp.obj 
/home/hooke007/mpv64/install/bin/x86_64-w64-mingw32-g++ -DALC_API="" -DAL_ALEXT_PROTOTYPES -DAL_API="" -DAL_BUILD_LIBRARY -DAL_LIBTYPE_STATIC -DNOMINMAX -DRESTRICT=__restrict -D_WIN32 -D__USE_MINGW_ANSI_STDIO -I/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/include -I/usr/include/pipewire-0.3 -I/usr/include/spa-0.2 -I/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft-build -I/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft -I/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/common -O2 -g -D_DEBUG -Winline -Wunused -Wall -Wextra -Wshadow -Wconversion -Wcast-align -Wpedantic -Wold-style-cast -Wnon-virtual-dtor -Woverloaded-virtual -fno-math-errno -std=gnu++14 -MD -MT CMakeFiles/OpenAL.dir/alc/backends/pipewire.cpp.obj -MF CMakeFiles/OpenAL.dir/alc/backends/pipewire.cpp.obj.d -o CMakeFiles/OpenAL.dir/alc/backends/pipewire.cpp.obj -c /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/backends/pipewire.cpp
In file included from /usr/include/pipewire-0.3/pipewire/protocol.h:48,
                 from /usr/include/pipewire-0.3/pipewire/proxy.h:108,
                 from /usr/include/pipewire-0.3/pipewire/client.h:35,
                 from /usr/include/pipewire-0.3/pipewire/pipewire.h:35,
                 from /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/backends/pipewire.cpp:54:
/usr/include/pipewire-0.3/pipewire/utils.h:33:10: fatal error: sys/un.h: No such file or directory
   33 | #include <sys/un.h>
      |          ^~~~~~~~~~
compilation terminated.
[53/95] Building CXX object CMakeFiles/OpenAL.dir/alc/alc.cpp.obj
In file included from /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/alconfig.h:6,
                 from /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/device.h:14,
                 from /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/al/auxeffectslot.h:11,
                 from /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/alc.cpp:62:
/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/common/aloptional.h: In function 'ALCenum {anonymous}::UpdateDeviceParams(ALCdevice*, const int*)':
/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/common/aloptional.h:313:75: warning: 'optlimit.al::optional<bool>::mStore.al::detail_::optional_storage<bool, true, true, true, true>::<anonymous>.al::detail_::optstore_helper<bool>::<anonymous>.al::detail_::optstore_base<bool, true>::<anonymous>.al::detail_::optstore_base<bool, true>::<unnamed union>::mDummy' may be used uninitialized in this function [-Wmaybe-uninitialized]
  313 |     { return bool{*this} ? **this : static_cast<T>(std::forward<U>(defval)); }
      |                                                                           ^
/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/alc.cpp:1536:24: note: 'optlimit.al::optional<bool>::mStore.al::detail_::optional_storage<bool, true, true, true, true>::<anonymous>.al::detail_::optstore_helper<bool>::<anonymous>.al::detail_::optstore_base<bool, true>::<anonymous>.al::detail_::optstore_base<bool, true>::<unnamed union>::mDummy' was declared here
 1536 |     al::optional<bool> optlimit{};
      |                        ^~~~~~~~
/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/alc.cpp:1693:13: warning: '*((void*)& optchans +1)' may be used uninitialized in this function [-Wmaybe-uninitialized]
 1693 |             if(device->FmtChans == DevFmtAmbi3D)
      |             ^~
/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/alc.cpp:1696:39: warning: '*((void*)& optlayout +1)' may be used uninitialized in this function [-Wmaybe-uninitialized]
 1696 |                 device->mAmbiLayout = *optlayout;
      |                                       ^~~~~~~~~~
/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/alc.cpp:1697:36: warning: '*((void*)& optscale +1)' may be used uninitialized in this function [-Wmaybe-uninitialized]
 1697 |                 device->mAmbiScale = *optscale;
      |                 ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/alc.cpp:1686:26: warning: '*((void*)& opthrtf +1)' may be used uninitialized in this function [-Wmaybe-uninitialized]
 1686 |             stereomode = *opthrtf ? StereoEncoding::Hrtf : StereoEncoding::Normal;
      |                          ^~~~~~~~
[54/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/pshifter.cpp.obj
[55/95] Building CXX object CMakeFiles/OpenAL.dir/alc/backends/loopback.cpp.obj
[56/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/vmorpher.cpp.obj
[57/95] Building CXX object CMakeFiles/OpenAL.dir/alc/backends/base.cpp.obj
[58/95] Building CXX object CMakeFiles/OpenAL.dir/alc/backends/null.cpp.obj
[59/95] Building CXX object CMakeFiles/OpenAL.dir/alc/effects/reverb.cpp.obj
[60/95] Building CXX object CMakeFiles/OpenAL.dir/alc/panning.cpp.obj
In file included from /home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/panning.cpp:50:
/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/core/ambdec.h: In function 'void aluInitRenderer(ALCdevice*, int, al::optional<StereoEncoding>)':
/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/core/ambdec.h:18:8: warning: inlining failed in call to 'AmbDecConf::~AmbDecConf() noexcept': call is unlikely and code size would grow [-Winline]
   18 | struct AmbDecConf {
      |        ^~~~~~~~~~
/home/hooke007/mpv64/packages/openal-soft-prefix/src/openal-soft/alc/panning.cpp:875:29: note: called from here
  875 |             AmbDecConf conf{};
      |                             ^
[61/95] Building CXX object CMakeFiles/OpenAL.dir/alc/backends/winmm.cpp.obj
ninja: build stopped: subcommand failed.
```
